posthog/hogql_queries/insights/retention_query_runner.py

```diff
@@ -875,3 +873,76 @@ def to_events_query(
                 events_query.where = ast.And(exprs=[cast(ast.Expr, existing_where), ast.And(exprs=event_filters)])
 
             return events_query
+
+    def _get_return_event_timestamps_expr(
+        self, minimum_occurrences: int, start_of_interval_sql: Expr, return_entity_expr: Expr
+    ) -> Expr:
+        if minimum_occurrences > 1:
+            return parse_expr(
+                """
+                arrayFilter(
+                (date, counts) -> counts >= {minimum_occurrences},
+                date_range,
+                return_event_counts_by_interval,
```

Comment:
there's a weird dependency here that `return_event_counts_by_interval` is added when `minimum_occurrences > 1` in a different function. Lets put a comment here at least to document that